cmd/telemeter-server/main.go

```diff
@@ -242,30 +241,45 @@ func (o *Options) Run() error {
 			Timeout:   20 * time.Second,
 			Transport: telemeter_http.NewInstrumentedRoundTripper("authorize", transport),
 		}
+	}
 
-		if o.OIDCIssuer != "" {
-			provider, err := oidc.NewProvider(ctx, o.OIDCIssuer)
-			if err != nil {
-				return fmt.Errorf("OIDC provider initialization failed: %v", err)
-			}
+	forwardClient := http.Client{
+		Timeout: 5 * time.Second,
+		Transport: telemeter_http.NewInstrumentedRoundTripper("forward",
+			&http.Transport{
+				MaxIdleConnsPerHost: 10,
+				IdleConnTimeout:     30 * time.Second,
+				DialContext:         (&net.Dialer{Timeout: 10 * time.Second}).DialContext,
+			},
+		),
+	}
 
-			ctx = context.WithValue(ctx, oauth2.HTTPClient,
-				&http.Client{
-					Timeout:   20 * time.Second,
-					Transport: telemeter_http.NewInstrumentedRoundTripper("oauth", transport),
-				},
-			)
+	if o.OIDCIssuer != "" {
+		provider, err := oidc.NewProvider(ctx, o.OIDCIssuer)
+		if err != nil {
+			return fmt.Errorf("OIDC provider initialization failed: %v", err)
+		}
 
-			cfg := clientcredentials.Config{
-				ClientID:     o.ClientID,
-				ClientSecret: o.ClientSecret,
-				TokenURL:     provider.Endpoint().TokenURL,
-			}
+		ctx = context.WithValue(ctx, oauth2.HTTPClient,
+			&http.Client{
+				Timeout:   20 * time.Second,
+				Transport: telemeter_http.NewInstrumentedRoundTripper("oauth", authorizeClient.Transport),
+			},
+		)
+
+		cfg := clientcredentials.Config{
+			ClientID:     o.ClientID,
+			ClientSecret: o.ClientSecret,
+			TokenURL:     provider.Endpoint().TokenURL,
+		}
 
-			authorizeClient.Transport = &oauth2.Transport{
-				Base:   authorizeClient.Transport,
-				Source: cfg.TokenSource(ctx),
-			}
+		authorizeClient.Transport = &oauth2.Transport{
+			Base:   authorizeClient.Transport,
+			Source: cfg.TokenSource(ctx),
+		}
+		forwardClient.Transport = &oauth2.Transport{
+			Base:   authorizeClient.Transport,
```

Comment:
This should be `forwardClient.Transport`, othewise we are double wrapping a transport in OAuth2 roundtrippers